script.js

```diff
@@ -0,0 +1,959 @@
+        color: currentCardColor,
+        hold: false, // Indicates whether this card is on hold or not
+      };
+
+      // console.log(`rank: ${rankCounter}`);
+
+      // add the card to the deck
+      newDeck.push(card);
+    }
+  }
+  return newDeck;
+};
+
+// To check whether it's a pair of Jack or better
+// Pair of jacks or better: Two jacks, queens, kings, or aces.
+const isPairOfJacksOrBetter = () => {
+  // Testing purpose
+  // boardOfDealtCards = TestArray1;
+  /// ////////////////////////////
+
+  let hasPairFound = false;
+  for (let i = 0; i < boardOfDealtCards.length - 1; i += 1)
+  {
+    const currentCard = boardOfDealtCards[i];
+    // If the current cards display name is neither of A, J, K and Q, proceed with next card
+    if (!((currentCard.display === JACK_DISPLAY_NAME)
+    || (currentCard.display === KING_DISPLAY_NAME)
+    || (currentCard.display === QUEEN_DISPLAY_NAME)
+    || (currentCard.display === ACE_DISPLAY_NAME)))
+    {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+    // Compare with each other elements in the array
+    for (let j = i + 1; j < boardOfDealtCards.length; j += 1)
+    {
+      const nextCard = boardOfDealtCards[j];
+      // If atleast one pair is found, no further checking is done.
+      // It's confirmed as PairOfJacksOrBetter
+      if (currentCard.display === nextCard.display)
+      {
+        hasPairFound = true;
+        break;
+      }
+    }
+    if (hasPairFound)
+    {
+      break;
+    }
+  }
+  return hasPairFound;
+};
+
+// Is Two Pair - Two cards of one rank, two cards of another rank;
+// for example, ace of spades, ace of hearts, 7 of clubs, 7 of diamonds.
+const isTwoPair = () => {
+  let numOfPairsFound = 0;
+  const totalCardCount = boardOfDealtCards.length;
+  // Variable to store the indices that are already a part of a pair
+  const isPairFoundArray = [];
+  for (let idx = 0; idx < totalCardCount; idx += 1)
+  {
+    isPairFoundArray.push(false);
+  }
+  for (let i = 0; i < totalCardCount - 1; i += 1)
+  {
+    // If the this card is already a part of pair, skip to next
+    if (isPairFoundArray[i])
+    {
+      continue;
+    }
+    const currentCard = boardOfDealtCards[i];
+    // Compare with each other elements in the array
+    for (let j = i + 1; j < totalCardCount; j += 1)
+    {
+      // If the this card is already a part of pair, skip to next
+      if (isPairFoundArray[j])
+      {
+        continue;
+      }
+      const nextCard = boardOfDealtCards[j];
+      if (currentCard.rank === nextCard.rank)
+      {
+        numOfPairsFound += 1;
+        isPairFoundArray[i] = true;
+        isPairFoundArray[j] = true;
+      }
+    }
+  }
+
+  console.log(`boardOfDealtCards: ${boardOfDealtCards}`);
+  console.log(`numOfPairsFound: ${numOfPairsFound}`);
+  console.log(`isPairFoundArray: ${isPairFoundArray}`);
+  // If 2 pairs are found, return true;
+  if (numOfPairsFound >= 2)
+  {
+    return true;
+  }
+  return false;
```

Comment:
This logic can be simplified to `return numOfPairsFound >= 2;`

Reply:
Updated. That was weird. :)